### PR TITLE
Included the first "Today I Learned" project for the command line

### PR DESCRIPTION
**Pull request by Dangeranger**
This project is intended to integrate the knowledge students have acquired
regarding NodeJS, JavaScript, Terminal commands, command line flags,
Input/Output, formatting strings, and Date-Times.

The stories are designed to challenge a student to reach beyond what they have
learned at that point, and force them to connect several concepts together in
order to produce a coherent application.

### CODE REVIEW

File: public/projects/today_i_learned_cli.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+#### Given ####
+An empty command prompt
+
+#### When ####
+
+A user inputs:
+
+``` sh
+til -n convert-json-to-javascript -t javascript -c "To convert JSON to a Javascript object use JSON.parse(myObject)"
```
Comment (alexch) [inline public/projects/today_i_learned_cli.md]: I think this is too many options for a first feature. Let's add categories later (default to 'general') and I don't see the value in naming entries; a TIL journal is just a list of notes, and it should be as frictionless as possible e.g. `til all numbers in JS are floats` 💥

File: public/projects/today_i_learned_cli.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+The TIL entry has a unique name of:
+
+``` text
+convert-json-to-javascript
+```
+
+#### And ###
+
+The TIL entry has a Date and Time in the format:
```
Comment (alexch) [inline public/projects/today_i_learned_cli.md]: This "has" requirement does not describe behavior. What does "has" mean? How do you verify it? Is there a file somewhere?

File: public/projects/today_i_learned_cli.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+Your task for this project is to build a command line program which you can use to record, search, edit, and reference answers for various programming questions and answers.
+
+## Wire Frames
+
+<!-- Maybe wire frames are not going to be helpful for the CLI version -->
+[TBD]()
+
+## Considerations
+
+* How should you ensure each TIL entry should has a unique name?
```
Comment (alexch) [inline public/projects/today_i_learned_cli.md]: typo: `should has` -> has

**Comment by Dangeranger** [comment]
@alexch Your suggested changes have been applied, as well as some additional cleanup. 

The PR has been rebased on top of master to make reviewing the diff with the current project easier.

**Pull request merged**